### PR TITLE
Switch to AWS SDK v2

### PR DESCRIPTION
**Pull request by bjwschaap**
As discussed in #886 this is a first PR migrating the s3 tier 2 storage uploader from AWS SDK v1 to v2.

### CODE REVIEW

**Comment by idanasulinmemphis** [comment]
@bjwschaap I'm on it

**Comment by yanivbh1** [comment]
Adding @Avitaltrifsik

**Review by idanasulinmemphis — COMMENTED** [review]
Thanks @cbnsndwch, it looks very good besides 1 comment I left.
I'm waiting for your reply

File: go.mod [hunk 1]
```diff
@@ -4,38 +4,43 @@ go 1.19
 
 require (
 	github.com/gin-contrib/cors v1.4.0
-	github.com/gin-gonic/contrib v0.0.0-20201101042839-6a891bf89f19
-	github.com/gin-gonic/gin v1.8.1
-	github.com/go-playground/validator/v10 v10.10.0
-	github.com/gofrs/uuid v4.2.0+incompatible
-	github.com/jhump/protoreflect v1.13.0
-	github.com/klauspost/compress v1.16.0
+	github.com/gin-gonic/contrib v0.0.0-20221130124618-7e01895a63f2
```
[the remark this answers is not in the document]
Comment (bjwschaap) [reply go.mod]: I've reverted all module upgrades not related to AWS SDK v2.

**Review by idanasulinmemphis — APPROVED** [review]
@bjwschaap thanks now it looks perfect, approved and merged.
It will be released on our next version.

**Pull request merged**